experimental/auth-logic/provenance_wrapper_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authlogic contains logic and tests for interfacing with the
+// authorization logic compiler
+package authlogic
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/project-oak/transparent-release/slsa"
+)
+
+func (p provenanceWrapper) Identify() Principal {
+	provenance, provenanceErr := slsa.ParseProvenanceFile(p.filePath)
+	if provenanceErr != nil {
+		panic(provenanceErr)
+	}
+
+	applicationName := provenance.Subject[0].Name
+	return Principal{fmt.Sprintf("%v::Provenance", applicationName)}
+}
+
+func TestProvenanceWrapper(t *testing.T) {
+	want := "oak_functions_loader::Provenance says {\nexpected_hash(oak_functions_loader::Binary, 15dc16c42a4ac9ed77f337a4a3065a63e444c29c18c8cf69d6a6b4ae678dca5c).\n}"
```

Comment:
It would be nice to always prefix hashes with the hash type (in this case `sha256:`), to avoid ambiguities, and also to allow mixing different hash types (e.g. git still uses sha1).

Reply:
IIRC, in the provenance files, the digest is under a different field, named after the hash type (e.g. sha256 in this case), but that information was being discarded. Now it is correctly propagated, thanks!